sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/FieldDeclarationsCountCheck.java

```diff
@@ -0,0 +1,91 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Checks that classes have at most 1 field of the given className.
+ *
+ * <p>This can be useful for example to ensure that only one Logger is used.
+ *
+ * @author Milos Fabian, Pantheon Technologies - original author (in OpenDaylight)
+ * @author Michael Vorburger.ch - refactored and made more generalized for contribution to Sevntu
+ */
+public class FieldDeclarationsCountCheck extends AbstractCheck {
+
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "field.count";
+
+    /**
+     * Class name check for.
+     */
+    private String className;
+
+    /**
+     * Field to hold previously seen Logger.
+     */
+    private String prevSeenClassName = "";
+
+    /**
+     * Set Class of which there should only be max. 1 field per class.
+     * @param className the fully qualified name (FQN) of the class
+     */
+    public void setClassName(String className) {
+        this.className = className;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {TokenTypes.VARIABLE_DEF };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isAFieldVariable(ast) && Utils.isMatchingClass(ast, className)) {
```

Comment:
keep thinking like this on my each future requests :) , I know I over-demanding for quality.